src/Providers/Stripe/Card/Service.php

```diff
@@ -0,0 +1,185 @@
+        }
+    }
+
+    /**
+     * Tries create a card using the Stripe SDK.
+     *
+     * @param Token $token
+     * @param array $params
+     *
+     * @return Card
+     *
+     * @throws Exception
+     * @throws CardException
+     */
+    public function create(Token $token, array $params = []): Card
+    {
+        switch ($token->type) {
+            case StripeToken::UNDEFINED:
+                $params = array_merge(
+                    [
+                        'usage' => 'on_session',
+                        'payment_method_types' => ['card'],
+                    ],
+                    $params
+                );
+
+                $response = SetupIntent::create($params);
+
+                break;
+            case StripeToken::SETUP_INTENT:
+                $response = SetupIntent::retrieve($token->value);
+
+                break;
+            default:
+                throw new \Exception('You either need to pass a null Token or a setup intent one.');
+        }
+
+        if ($response->status !== SetupIntent::STATUS_SUCCEEDED) {
+            $token = new StripeToken($response->client_secret);
+
+            throw new CardException($token);
+        }
+
+        /** @var PaymentMethod $paymentMethod */
+        $paymentMethod = PaymentMethod::retrieve($response->payment_method);
+
+        return new Card(
+            $paymentMethod->id,
+            $paymentMethod->card->last4,
+            $paymentMethod->card->brand,
+            $paymentMethod->card->funding,
+            $paymentMethod->card->exp_month,
+            $paymentMethod->card->exp_year
+        );
+    }
+
+    /**
+     * Tries get cards using the Stripe SDK.
+     *
+     * @param Token $token
+     *
+     * @return array[Card]
+     */
+    public function index(Token $token): array
+    {
+        $cards = [];
+
+        switch ($token->type)
+        {
+            case StripeToken::CUSTOMER:
+                $paymentMethods = PaymentMethod::all([
+                    "customer" => $token->value,
+                    "type" => "card",
+                ]);
+
+                if (isset($paymentMethods->data) && $paymentMethods->data) {
+                    foreach ($paymentMethods->data as $paymentMethod) {
+                        $cards[] = new Card(
+                            $paymentMethod->id,
+                            $paymentMethod->card->last4,
+                            $paymentMethod->card->brand,
+                            $paymentMethod->card->funding,
+                            $paymentMethod->card->exp_month,
+                            $paymentMethod->card->exp_year
+                        );
+                    }
+                }
+
+                /** @var StripeCustomer $customer */
+                $customer = StripeCustomer::retrieve($token->value);
+
+                if (isset($customer->cards->data) && $customer->cards->data) {
```

Comment:
Is this for backwards compatibility?
Looking at docs I cannot see cards on Customer object, unless I am looking in the wrong place ;).
https://stripe.com/docs/api/customers/retrieve

Reply:
Yeah it's how we return cards from the sources API. 
See: https://github.com/Pod-Point/api3/blob/9a7da51c0976519df13906bc17c8ba006acad78a/app/Services/StripePaymentHandler.php#L184